pym/bob/input.py

```diff
@@ -2224,6 +2225,23 @@ def validate(self, data):
 
 class RecipeSet:
 
+    BUILD_DEV_SCHEMA = schema.Schema(
+        {
+            schema.Optional('destination') : str,
+            schema.Optional('force') : bool,
+            schema.Optional('no-deps') : bool,
+            schema.Optional('build-only') : bool,
+            schema.Optional('checkout-only') : bool,
```

Comment:
`--build-only` and `--checkout-only` are mutually exclusive. Maybe `build-mode` with `normal`, `build-only` and `checkout-only` as possible values?